src/test/groovy/com/gradle/BaseInitScriptTest.groovy

```diff
@@ -281,4 +295,96 @@ abstract class BaseInitScriptTest extends Specification {
             return "Gradle " + gradleVersion.version
         }
     }
+
+    static TestDvPluginVersion dvPlugin(DvPluginId id, String version, boolean deprecated = false) {
+        return new TestDvPluginVersion(id, version, deprecated)
+    }
+
+    static enum DvPluginId {
+        DEVELOCITY('com.gradle.develocity'),
+        GRADLE_ENTERPRISE('com.gradle.enterprise'),
+        BUILD_SCAN('com.gradle.build-scan');
+        final String id;
+
+        DvPluginId(String id) {
+            this.id = id
+        }
+    }
+
+    static final class TestDvPluginVersion {
+        final DvPluginId pluginId
+        final String version
+        final boolean deprecated
+
+        TestDvPluginVersion(DvPluginId pluginId, String version, boolean deprecated) {
+            this.pluginId = pluginId
+            this.version = version
+            this.deprecated = deprecated
+        }
+
+        String getId() {
+            return pluginId.id
+        }
+
+        boolean isCompatibleWith(TestGradleVersion gradleVersion) {
+            switch (pluginId) {
```

Comment:
I'm not sure what you mean. I intentionally kept all comparisons using `<` and `<=` for ease of reading. Can you provide an example of what you're suggesting?

Reply:
Yes, I mean:

```groovy
return gradleVersion.gradleVersion >= GRADLE_5 && gradleVersion.gradleVersion < GRADLE_6
```

Instead of:

```groovy
return GRADLE_5 <= gradleVersion.gradleVersion && gradleVersion.gradleVersion < GRADLE_6
```

I see how you have it written is consistent, from a certain point of view. Since it really comes down to preference we can resolve this.